File: src/dataset/python/acronym_exercism-ThomasZumsteg/claude/ClaudeSonnet4_acronym_v4.py
from re import split

def abbreviate(phrase):
    """abbreviate creates an acronym for a phrase"""
    result = []
    for word in split(r'\W+', phrase):
        if word:
            if word.isupper():
                result.append(word[0])
            else:
                result.append(word[0].upper())
                result.extend(char for char in word[1:] if char.isupper())
    return ''.join(result)

def abbreviate_word(word):
    """abbreviate_word selects the letters in a word to use for abbriviation"""
    if word.isupper():
        return word[0]
    return word[0].upper() + ''.join(letter for letter in word[1:] if letter.isupper())

File: src/dataset/typescript/acronym_exercism-wackerow/claude/ClaudeSonnet4_acronym_v4.ts
export default class Acronym {
  public static parse(phrase: string): string {
    const result: string[] = [];
    const length = phrase.length;
    let prevChar = '';
    
    for (let i = 0; i < length; i++) {
      const char = phrase[i];
      
      if (char === ':') break;
      
      if (char >= 'A' && char <= 'Z') {
        result.push(char);
      } else if ((char >= 'a' && char <= 'z') && (prevChar === ' ' || prevChar === '-')) {
        result.push(String.fromCharCode(char.charCodeAt(0) - 32));
      }
      
      prevChar = char;
    }
    
    return result.join('');
  }
}